libsolidity/ast/Types.cpp

```diff
@@ -1523,8 +1528,6 @@ TypePointer ArrayType::interfaceType(bool _inLibrary) const
 	TypePointer baseExt = m_baseType->interfaceType(_inLibrary);
 	if (!baseExt)
 		return TypePointer();
-	if (m_baseType->category() == Category::Array && m_baseType->isDynamicallySized())
```

Comment:
So before this change dynamic arrays were disallowed, after this they are allowed. I guess there were no tests checking for this (since the test output didn't change)?